source/common/config/xds_manager_impl.h

```diff
@@ -10,20 +9,24 @@ namespace Config {
 
 class XdsManagerImpl : public XdsManager {
 public:
-  XdsManagerImpl(Upstream::ClusterManager& cm,
-                 ProtobufMessage::ValidationContext& validation_context)
-      : cm_(cm), validation_context_(validation_context) {}
+  XdsManagerImpl(ProtobufMessage::ValidationContext& validation_context)
+      : validation_context_(validation_context) {}
 
   // Config::ConfigSourceProvider
+  absl::Status initialize(Upstream::ClusterManager* cm) override;
+  void shutdown() override {}
```

Comment:
will this be implemented later?

Reply:
Yes, this will shutdown the ADS once ads_mux_ is moved from the cluster-manager to the xds-manager (see [here](https://github.com/envoyproxy/envoy/blob/6dde5284615e10b2f36211e20b673009cd33b410/source/common/upstream/cluster_manager_impl.h#L298).
The reason I'm introducing the method at this stage is to reduce the amount of changes in the next refactors (smaller PRs).